source/gloperate/include/gloperate/pipeline/Slot.inl

```diff
@@ -183,10 +188,26 @@ void Slot<T>::setValid(bool isValid)
     }
 
     // Set state of own data
+    auto wasValid = m_valid;
     m_valid = isValid;
 
-    // Emit signal
-    this->onValueChanged(this->m_value);
+    // Emit signal if it was invalidated
+    if(wasValid && !isValid)
```

Comment:
Does this conform to #310. Invalidation is signaled (including propagation) when the slot changes from valid to invalid; or when the connected source invalidates (see Slot::connect). That means setting a Slot to valid can never trigger the signal nor can an invalid Slot signal again.